dissect/target/plugins/os/default/network.py

```diff
@@ -72,26 +75,22 @@ def dns(self) -> list[str | IPAddress]:
 
     @internal
     def with_ip(self, ip_addr: str) -> Iterator[InterfaceRecord]:
+        """Yield all InterfaceRecords that use ``ip_addr``."""
         for interface in self.interfaces():
-            if ip_addr in interface.ip:
+            if any(iface.ip == ip_addr for iface in interface.interface):
```

Comment:
Am I correct in seeing that this basically enumerates all IP addresses in the network range? Surely there's a better API?